### DISCUSSION TITLE
Scroll Bars with RDK

### DISCUSSION THREAD

**Discussion by wsr017** [jspsych/jsPsych#787] [Q&A] [answered]
Hello everyone,

I have a question regarding the RDK plugin. When I run the plugin and look at the resulting dot motion field, I always have vertical and horizontal scroll bars and can move the screen around a little using the arrow keys (as long as the arrow keys are not the specified keyboard response choices). In jspsych.css, I have overflow-y: auto as was recommended in another post, but the scrolling remains. 

Is there a way to remove scrolling on the RDK plugin, maybe by specifying the size of the display so there is no overflow? 

Thanks so much!

**Comment by vijaymarupudi** [comment, marked as answer]
I'm not familiar with the plugin, but these edits seem to help

```diff
diff --git a/plugins/jspsych-rdk.js b/plugins/jspsych-rdk.js
index 40febc5..5064ff8 100644
--- a/plugins/jspsych-rdk.js
+++ b/plugins/jspsych-rdk.js
@@ -372,6 +372,9 @@ jsPsych.plugins["rdk"] = (function() {
 		//Remove the margins and padding of the canvas
 		canvas.style.margin = 0;
 		canvas.style.padding = 0;		
+    canvas.style.position = 'absolute';
+    canvas.style.top = 0
+    canvas.style.left = 0;
 		
 		//Get the context of the canvas so that it can be painted on.
 		var ctx = canvas.getContext("2d");

```

**Reply by jodeleeuw** [reply]
Elevating this to an issue so we can implement a fix.

**Reply by gorkemer** [reply]
Hello,

I'd like to note here that having canvas.style.position = 'absolute' in the RDK plugin somehow disrupts the scaling of canvas done by the "resize" plugin.

**Reply by gorkemer** [reply]
Alternatively, `body.style.overflowY = "hidden"`can be added to the RDK plugin to hide the scroll bar in the display element. This way the resizing by the plugin wouldn't mess the canvas size created by the RDK plugin.

**Reply by vijaymarupudi** [reply]
@jodeleeuw (for the fix) and @gorkemer I think the correct fix here is to use `canvas.style.display = 'block'` because the canvas element is an inline element by default, instead of my duct tape fix mentioned earlier.

**Reply by becky-gilbert** [reply]
Thanks @vijaymarupudi and @gorkemer! I created a new issue for this so that we can track and fix it: #1536 Please feel free to add any other info and suggestions there.
